Community/Tdarr_Plugin_vdka_Tiered_CPU_CRF_Based_Configurable.js

```diff
@@ -0,0 +1,150 @@
+       the CRF value is configurable per resolution.
+       FFmpeg Preset can be configured, uses slow by default. 
+       ALL OPTIONS MUST BE CONFIGURED UNLESS MARKED OPTIONAL!
+       If files are not in hevc they will be transcoded. 
+       The output container is mkv. \n\n`,
+    Version: '1.00',
+    Link:
+        'https://github.com/HaveAGitGat/Tdarr_Plugins/blob/master/Community/'
+      + ' Tdarr_Plugin_vdka_Tiered_CPU_CRF_Based_Configurable.js',
+    Tags: 'pre-processing,ffmpeg,video only,h265,configurable',
+
+    Inputs: [
+      {
+        name: 'sdCRF',
+        tooltip: `Enter the CRF value you want for 480p and 576p content. (0-51, lower = higher quality, bigger file)
+         \\nExample:\\n 
+        
+        19`,
+      },
+      {
+        name: 'hdCRF',
+        tooltip: `Enter the CRF value you want for 720p content. (0-51, lower = higher quality, bigger file)
+        
+        \\nExample:\\n
+        21`,
+      },
+      {
+        name: 'fullhdCRF',
+        tooltip: `Enter the CRF value you want for 1080p content. (0-51, lower = higher quality, bigger file)
+        
+        \\nExample:\\n
+        23`,
+      },
+      {
+        name: 'uhdCRF',
+        tooltip: `Enter the CRF value you want for 4K/UHD/2160p content. (0-51, lower = higher quality, bigger file)
+        
+        \\nExample:\\n
+        26`,
+      },
+      {
+        name: 'bframe',
+        tooltip: `Specify amount of b-frames to use, 0-16.
+        
+        \\nExample:\\n
+        8`,
+      },
+      {
+        name: 'ffmpegPreset',
+        tooltip: `OPTIONAL, DEFAULTS TO SLOW IF NOT SET 
+        \\n Enter the ffmpeg preset you want, leave blank for default (slow) 
+        
+        \\nExample:\\n 
+          slow  
+        
+        \\nExample:\\n 
+          medium  
+        
+        \\nExample:\\n 
+          fast  
+        
+        \\nExample:\\n 
+          veryfast`,
+      },
+    ],
+  };
+}
+
+module.exports.plugin = function plugin(file, librarySettings, inputs) {
+  let crf = '';
+  // default values that will be returned
+  const response = {
+    processFile: true,
+    preset: '',
+    container: '.mkv',
+    handBrakeMode: false,
+    FFmpegMode: true,
+    reQueueAfter: true,
+    infoLog: '',
+  };
+
+  // check if the file is a video, if not the function will be stopped immediately
+  if (file.fileMedium !== 'video') {
+    response.processFile = false;
+    response.infoLog += '☒File is not a video! \n';
+    return response;
+  }
+  response.infoLog += '☑File is a video! \n';
+
+  // check if the file is already hevc
+  // it will not be transcoded if true and the plugin will be stopped immediately
+  for (let i = 0; i < file.ffProbeData.streams.length; i += 1) {
+    if (file.ffProbeData.streams[i].codec_name.toLowerCase() === 'hevc') {
+      response.processFile = false;
+      response.infoLog += '☑File is already in hevc! \n';
+      break;
+    }
+  }
```

Comment:
Do we need this to be a loop, if the file is a video it's safe to assume the first stream is a video stream.

Reply:
Better safe than sorry, this definitely stays